Generals/Code/GameEngine/Source/GameLogic/Object/Behavior/PropagandaTowerBehavior.cpp

```diff
@@ -205,11 +206,25 @@ UpdateSleepTime PropagandaTowerBehavior::update( void )
 		}
 	}
 
-	if( self->getContainedBy()  &&  self->getContainedBy()->getContainedBy() )
+#if RETAIL_COMPATIBLE_CRC
+	Bool contained = self->getContainedBy() && self->getContainedBy()->getContainedBy();
+#else
+	Bool contained = false;
+
+	for (Object* child = self, *container = self->getContainedBy(); container; child = container, container = container->getContainedBy())
+	{
+		ContainModuleInterface* containModule = container->getContain();
+		if (containModule && containModule->isEnclosingContainerFor(child))
+		{
+			contained = true;
+			break;
+		}
+	}
+#endif
+
+	if (contained)
 	{
-		// If our container is contained, we turn the heck off.  Seems like a weird specific check, but all of
-		// attacking is guarded by the same check in isPassengersAllowedToFire.  We similarly work in a container,
-		// but not in a double container.
+		// If our container or any parent containers are an enclosing container, we turn the heck off.
```

Comment:
I suggest

`// TheSuperHackers @bugfix If our container or any parent containers are enclosing, we turn the heck off.`

and move comment below `#else` ?